pkg/controller/v1beta1/inferenceservice/components/explainer.go

```diff
@@ -88,10 +88,26 @@ func (e *Explainer) Reconcile(isvc *v1beta1.InferenceService) error {
 		addAgentContainerPort(&isvc.Spec.Explainer.PodSpec.Containers[0])
 	}
 
+	deployConfig, err := v1beta1.NewDeployConfig(e.client)
+	if err != nil {
+		return err
+	}
+
 	podSpec := v1.PodSpec(isvc.Spec.Explainer.PodSpec)
 
 	// Here we allow switch between knative and vanilla deployment
-	if value, ok := annotations[constants.RawDeploymentAnnotationKey]; ok && value == "true" {
+	/*
+		add an item in inferenceservice configmap, called deploy.
+		case 1:
+		if deploy.DefaultDeploymentMode is Serverless, that means the default deployment mode is serverless(or knative),
+		if you want raw deployment, add an annotation "SwitchDeploymentMode" with value "true"
```

Comment:
I think `switchDeploymentMode` might not be clear for what mode you are deploying, how about just be explicit for the `DeploymentMode`, if it is `serverless` then deploy knave otherwise deploy `raw`. If knative is not installed then `serverless` is not supported.

Reply:
OK